packages/api-bindings/src/apollo/TypePolicy.ts

```diff
@@ -15,16 +15,23 @@ type TypedFieldFunctionOptions<TAll> = Overwrite<
   }
 >;
 
-type TypedFieldReadFunction<TExisting, TAll> = (
+export type FieldReadFunction<TExisting, TAll> = (
   existing: Readonly<TExisting> | undefined,
-  options: TypedFieldFunctionOptions<TAll>,
+  options: FieldFunctionOptions<TAll>,
 ) => TExisting | undefined;
 
+export type FieldPolicy<TExisting, TAll> = Overwrite<
+  UnpatchedFieldPolicy<TExisting>,
+  { read?: FieldReadFunction<TExisting, TAll> }
+>;
+
+type FieldPolicies<T> = Partial<{
+  [key in keyof T]: FieldPolicy<T[key], T> | FieldReadFunction<T[key], T>;
```

Comment:
Previously if the field policy was implemented as the object:

```ts
{
  read(...) {...},
  merge(...) {...}
}
```
The read function was poorly typed by the original Apollo `FieldReadFunction<any>`. While I was fixing the types here I aligned this other syntax for field policies to take benefit of our patched `FieldReadFunction` implementation.